install.sh

```diff
@@ -303,10 +303,18 @@ nvm_check_global_modules() {
 }
 
 nvm_do_install() {
-  if [ -n "${NVM_DIR-}" ] && ! [ -d "${NVM_DIR}" ]; then
-    echo >&2 "You have \$NVM_DIR set to \"${NVM_DIR}\", but that directory does not exist. Check your profile files and environment."
+  # If NVM_DIR is set to something other than the default, exit
```

Comment:
This doesn't seem like the right behavior - it's expected and by design that a user setting `NVM_DIR` beforehand can alter the location in which nvm is installed.

The specific thing that's a problem is when the specified override directory does not *exist* (and this PR is, I believe, hoping to skip that error checking when the directory doesn't exist but it's the default path)

Reply:
> If NVM_dir is defined and it is NOT what the installer would set it to (i.e. "$HOME/.nvm"), complain and exit

https://github.com/creationix/nvm/issues/1962#issuecomment-450133106
This comment in the issue had the above suggestion and I assumed this is the required behavior.

So what would the correct behavior be if I ignore that comment? Should I attempt to create the specified override directory if it does not exist?

Reply:
The only time I think it's reasonable to create the directory is if it a) does not exist and b) is what the default would be anyways. If it does not exist, and is *not* the default, I think the current behavior should remain.